Library/Homebrew/tab.rb

```diff
@@ -31,7 +31,12 @@ def self.create(formula, compiler, stdlib)
       "source" => {
         "path" => formula.path.to_s,
         "tap" => formula.tap ? formula.tap.name : nil,
-        "spec" => formula.active_spec_sym.to_s
+        "spec" => formula.active_spec_sym.to_s,
+        "versions" => {
+          "stable" => formula.stable ? formula.stable.version.to_s : nil,
+          "devel" => formula.devel ? formula.devel.version.to_s : nil,
+          "head" => formula.head ? formula.head.version.to_s : nil
+        }
```

Comment:
I think in addition to this you will also need to populate `attributes["source"]["versions"]` in `Tab.from_file_content`. Otherwise `Tab#versions` will return `nil` for existing tabs that were created before this change and this will cause `Tab#stable_version` and `Tab#devel_version` to fail.